packages/mdl-theme/_functions.scss

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import "./constants";
+
+/* All variables are local, so disable pattern checking. */
+/* stylelint-disable scss/dollar-variable-pattern */
+
+/*
+  Calculate the luminance for a color.
+  See https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-tests
+*/
+@function mdl-theme-luminance($color) {
+  $red: nth($mdl-theme-linear-channel-values, red($color) + 1);
+  $green: nth($mdl-theme-linear-channel-values, green($color) + 1);
+  $blue: nth($mdl-theme-linear-channel-values, blue($color) + 1);
+
+  @return .2126 * $red + .7152 * $green + .0722 * $blue;
+}
+
+/*
+  Calculate the contrast ratio between two colors.
+  See https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-tests
+*/
+@function mdl-theme-contrast($back, $front) {
+  $backLum: mdl-theme-luminance($back) + .05;
+  $foreLum: mdl-theme-luminance($front) + .05;
+
+  @return max($backLum, $foreLum) / min($backLum, $foreLum);
+}
+
+/* Determine whether to use dark or light text on top of color. */
+@function mdl-theme-light-or-dark($color) {
+  $minimumContrast: 3.1;
+
+  $lightContrast: mdl-theme-contrast($color, white);
+  $darkContrast: mdl-theme-contrast($color, rgba(black, .87));
+
+  @if ($lightContrast < $minimumContrast) and ($darkContrast > $lightContrast) {
+    @return "dark";
+  }
+
+  @else {
+    @return "light";
+  }
+}
```

Comment:
You need to re-enable the dollar variable pattern check at the end of this file:

```css
/* stylelint-enable scss/dollar-variable-pattern */
```

Otherwise I believe it just stays off.